packages/deep-merge/src/deep-merge.ts

```diff
@@ -35,22 +45,43 @@ export function replaceBehaviour<T extends object>(obj: T): T {
  * behavior.
  *
  * @param obj - The object to set its merge behavior.
+ *
+ * @returns The same object with the new behaviour.
  */
-export function deepMergeBehaviour<T extends object>(obj: T): T {
+export function deepMergeBehaviour<T extends Record<string, unknown>>(
+  obj: T
+): T {
   behaviourMap.set(obj, Behaviour.DeepMerge);
   return obj;
 }
 
-enum Behaviour { Replace = 'replace', DeepMerge = 'deep-merge' }
-
-function cloneSourcesProperties(target: any, source: any): (source: any) => void {
+/**
+ * A function which clones the properties of two sources.
+ *
+ * @param target - The target object to clone in.
+ * @param source - The source objet to clone.
+ *
+ * @returns The union of the target and source.
+ */
+function cloneSourcesProperties(
+  target: any,
+  source: any
+): (source: any) => void {
   if (source) {
     return Object.entries(source).reduce(cloneObjectProperties, target);
   } else {
     return target || {};
   }
 }
 
+/**
+ * A function which clones the properties of two object.
+ *
+ * @param target - The target object to clone in.
+ * @param source - The source objet to clone.
```

Comment:
This is not the source object, but an entry (key, value) to clone into the target object